File: src/main/java/net/sf/rails/game/state/StringChange.java
package net.sf.rails.game.state;

/**
 * Change associated with StringState
 * @author freystef
 */

final class StringChange extends Change {

    private final StringState state;
    private final String newValue;
    private final String oldValue;
    
    StringChange(StringState state, String newValue) {
        this.state = state;
        this.newValue = newValue;
        this.oldValue = state.value();
        super.init(state);
    }
    
    @Override void execute() {
        state.change(newValue);
    }

    @Override void undo() {
        state.change(oldValue);
    }

    @Override StringState getState() {
        return state;
    }

    @Override
    public String toString() {
        return "Change for " + state + ": From \"" + oldValue + "\" to \"" + newValue + "\""; 
    }
        
}

File: dotnet/RailsLib.Net/Net/Game/State/StringChange.cs
using System;
using System.Collections.Generic;
using System.Text;

/**
 * Change associated with StringState
 * @author freystef

 */
namespace GameLib.Net.Game.State
{
    public class StringChange : Change
    {
        private StringState state;
        private string newValue;
        private string oldValue;

        public StringChange(StringState state, string newValue)
        {
            this.state = state;
            this.newValue = newValue;
            this.oldValue = state.Value;
            base.Init(state);
        }

        override public void Execute()
        {
            state.Change(newValue);
        }

        override public void Undo()
        {
            state.Change(oldValue);
        }

        override public GameState GameState
        {
            get
            {
                return state;
            }
        }
        //override public StringState getState()
        //{
        //    return state;
        //}

        override public string ToString()
        {
            return "Change for " + state + ": From \"" + oldValue + "\" to \"" + newValue + "\"";
        }

    }
}
